test/extensions/filters/http/stateful_session/stateful_session_integration_test.cc

```diff
@@ -219,6 +233,151 @@ TEST_F(StatefulSessionIntegrationTest, NormalStatefulSession) {
   cleanupUpstreamAndDownstream();
 }
 
+TEST_F(StatefulSessionIntegrationTest, StatefulSessionHeaderWithPackagesAndStrictMode) {
+  initializeFilterAndRoute(
+      STATEFUL_SESSION_HEADER_FILTER_WITH_PACKAGES + STATEFUL_SESSION_STRICT_MODE, "");
+
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+
+  Http::TestRequestHeaderMapImpl request_headers{{":method", "GET"},
+                                                 {":path", "/test"},
+                                                 {":scheme", "http"},
+                                                 {":authority", "stateful.session.com"}};
+
+  auto response = codec_client_->makeRequestWithBody(request_headers, 0);
+
+  auto upstream_index = waitForNextUpstreamRequest({0, 1, 2, 3});
+  ASSERT(upstream_index.has_value());
+
+  envoy::config::endpoint::v3::LbEndpoint endpoint;
+  setUpstreamAddress(upstream_index.value(), endpoint);
+  const std::string address_string =
+      fmt::format("127.0.0.1:{}", endpoint.endpoint().address().socket_address().port_value());
+  const std::string encoded_address =
+      Envoy::Base64::encode(fmt::format("{};origin:{}", address_string, "abcdefg"));
+
+  default_response_headers_.addCopy(Http::LowerCaseString("session-header"), "abcdefg");
```

Comment:
I am trying to understand this test. So, the request does not contain `session-header`, but upstream returns `session-header` with value "abcdefg". Envoy will reply to the client with `session-header` equal to <upstreamIP>;origin:abcdefg. Correct?

Reply:
Yeah

Reply:
It is clear now. Thanks.